apps/meteor/ee/app/livechat-enterprise/server/api/contacts.ts

```diff
@@ -0,0 +1,91 @@
+import Ajv from 'ajv';
+
+import { API } from '../../../../../app/api/server';
+import { changeContactBlockStatus, closeBlockedRoom, hasSingleContactLicense } from './lib/contacts';
+
+const ajv = new Ajv({
+	coerceTypes: true,
+});
+
+type blockContactProps = {
+	contactId: string;
+	visitorId: string;
+};
+
+const blockContactSchema = {
+	type: 'object',
+	properties: {
+		contactId: {
+			type: 'string',
+		},
+		visitorId: {
+			type: 'string',
+		},
+	},
+	required: ['contactId', 'visitorId'],
+	additionalProperties: false,
+};
+
+const isBlockContactProps = ajv.compile<blockContactProps>(blockContactSchema);
+
+declare module '@rocket.chat/rest-typings' {
+	// eslint-disable-next-line @typescript-eslint/naming-convention
+	interface Endpoints {
+		'/v1/omnichannel/contacts.block': {
+			POST: (params: blockContactProps) => void;
+		};
+		'/v1/omnichannel/contacts.unblock': {
+			POST: (params: blockContactProps) => void;
+		};
+	}
+}
+
+API.v1.addRoute(
+	'omnichannel/contacts.block',
```

Comment:
i wonder if one endpoint with an `action` parameter would've been better. but this is a personal pref so if it was decided somewhere else to have both, ignore it.

Reply:
It's a good approach as well, but we're already aligned with FE

Reply:
no probs then!